Week 1 683380425-3 นายปฏิภาณ นามสีลี

Lab 1.1 Environment Setup & First Script

In [ ]:
print("Hello, Python Course!")

Hello, Python Course!


Lab 1.2 Working with Variables and Basic Data Types

2. Declare and print variables:

In [ ]:
Age = 19
GPA = 2.97
Favorite_Color = "Purple"
Student = True

print("Age = ",Age)
print("GPA = ",GPA)
print("Favorite_Color = ",Favorite_Color)
print("Student = ",Student)

Age =  19
GPA =  2.97
Favorite_Color =  Purple
Student =  True


3. Perform simple arithmetic:

In [ ]:
num1 = 123
num2 = 456

total_sum = num1 + num2
difference = num1 - num2
product = num1 * num2
division = num1 / num2

print("Sum:", total_sum)
print("Difference:", difference)
print("Product:", product)
print("Division:", division)

Sum: 579
Difference: -333
Product: 56088
Division: 0.26973684210526316


4. String concatenation and f-strings:

In [ ]:
First_name = "Patiphan"
Last_name = "Narmseelee"
Full_name = First_name + " " + Last_name
print(f"Hello,{Full_name}!")

Hello,Patiphan Narmseelee!


5. User input and type conversion:

In [ ]:
num = input("What is your favorite number : ")
print("type of input variable is : ", type(num))
num = int(num)
result = num + 5
print(num," + 5 = ", result)

What is your favorite number : 64
type of input variable is :  <class 'str'>
64  + 5 =  69


Activity 1.1 Based on the introduction, where do you think scripting with Python could be applied in real-world scenarios? Think about tasks you do daily or things you've seen in other courses

คำตอบ: ช่วยทำโปรเจกต์และวิชาคำนวณ ใช้คำนวณสูตรคณิตศาสตร์ สถิติ หรือแปลงหน่วยวิจัยยากๆ ในวิชาเรียนแทนการกดเครื่องคิดเลขทีละตัว

Activity 1.2:"Python Puzzlers" - Identifying Data Types & Output

1. สคริปต์สั้นๆ Word Problem

In [ ]:
celsius = float(input("ป้อนอุณหภูมิ (°C): "))

fahrenheit = (celsius * 9/5) + 32


print(f"{celsius}°C แปลงเป็นฟาเรนไฮต์ได้: {fahrenheit}°F")

ป้อนอุณหภูมิ (°C): 20
20.0°C แปลงเป็นฟาเรนไฮต์ได้: 68.0°F


In [ ]:
# Puzzler 1
x = 10
y = 3.5
z = x + y
print(z)
print(type(z))

13.5
<class 'float'>


Output: 13.5

<class 'float'>

DataType: Float

อธิบายว่าทำไมถึงคิดว่าได้ผลลัพธ์: โดย x เป็นจำนวนเต็ม (int) และ y เป็นทศนิยม (float) เมื่อนำมาบวกกัน Python จะแปลง int ให้กลายเป็น float โดยอัตโนมัติ เพื่อรักษาความละเอียดของทศนิยมไว้ ผลลัพธ์ z จึงออกมาเป็น float ($13.5$)

In [ ]:
# Puzzler 2
name = "Alice"
age = "30" # Note: This is a string
message = name + " is " + age + " years old."
print(message)
print(type(age))

Alice is 30 years old.
<class 'str'>


Output:Alice is 30 years old.

<class 'str'>

DataType: String

อธิบายว่าทำไมถึงคิดว่าได้ผลลัพธ์: เพราะตัวแปร age ถูกกำหนดค่าไว้ในเครื่องหมายคำพูด ("30") ตัวภาษา Python จึงมองว่ามันคือ ข้อความ (String)

In [ ]:
# Puzzler 3
is_sunny = True
is_raining = False
print(is_sunny and is_raining)
print(type(is_sunny))

False
<class 'bool'>


Output:False

<class 'bool'>



DataType: bool



อธิบายว่าทำไมถึงคิดว่าได้ผลลัพธ์: เพราะค่า True และ False (ต้องขึ้นต้นด้วยตัวพิมพ์ใหญ่ใน Python) คือค่าทางตรรกศาสตร์ ซึ่งจัดอยู่ในประเภท Boolean (bool)

In [ ]:
# Puzzler 4
# What happens here?
# num_str = input("Enter a number: ")
# result = num_str + 5
# print(result)

Output: เกิดข้อผิดพลาด (TypeError)

อธิบายว่าทำไมถึงคิดว่าได้ผลลัพธ์: เพราะฟังก์ชัน input() จะรับค่ากลับมาเป็นข้อความ (str) เสมอ แม้ว่าเราจะพิมพ์ตัวเลขลงไปก็ตาม เมื่อโปรแกรมพยายามนำข้อความมาบวก + กับตัวเลขตรงๆ โดยไม่มีการแปลงประเภทข้อมูลก่อน Python จะไม่สามารถประมวลผลได้และเกิด Error

